### PR TITLE
Replace use of cookies for token storage with localStorage

### PR DESCRIPTION
**Pull request by AlmightyYakob**
Depends on https://github.com/multinet-app/multinet-server/pull/472
Depends on https://github.com/multinet-app/multinetjs/pull/19.

Notable changes:

* Move login logic from `App.vue` to `LoginMenu.vue`
* Add method that checks the URL for an included loginToken, handling it if present.
* Use `userInfo` and `logout` functions of `multinetjs`
* Add interface for storing/retrieving login tokens through localStorage
* Fix incorrect encoding of `return_url` when logging in

### CODE REVIEW

File: src/utils/localStorage.ts [hunk 1]
```diff
@@ -0,0 +1,13 @@
+export const localStorageLoginTokenKey = 'loginToken';
```
Comment (waxlamp) [inline src/utils/localStorage.ts:1]: I don't think this needs to be exported.

File: src/api.ts [hunk 2]
```diff
@@ -1,24 +1,13 @@
...
-  });
-
-  if (resp.ok) {
-    return resp.json();
-  }
-
-  return null;
-}
+const storedLoginToken = getLoginToken();
+const api = multinetApi(`${host}/api`, storedLoginToken);
```
Comment (waxlamp) [inline src/api.ts:6]: You could just inline the call to `getLoginToken()`.

File: src/App.vue [hunk 3]
```diff
@@ -27,6 +30,33 @@ export default Vue.extend({
...
+    checkUrlForLogin() {
+      if (this.$route.hash) {
+        const result = this.$route.hash.match(loginTokenRegex);
+
+        if (result && result.length >= 2) {
+          const loginToken = result[1];
+          api.client.setAuthToken(loginToken);
+
+          const currentRouteName = this.$route.name;
+          const name = currentRouteName || 'home';
```
Comment (waxlamp) [inline src/App.vue]: Should inline `this.$route.name` into line 51.

**Review by waxlamp — APPROVED** [review]
Just a few minor code suggestions; looks good overall.

**Review by waxlamp — DISMISSED** [review]

**Comment by AlmightyYakob** [comment]
Thanks @waxlamp. Once all three of these PRs are approved, we can cut a release for multinetJS, make sure this passes CI, and then merge all at once.

File: src/components/LoginMenu.vue [hunk 4]
```diff
@@ -135,18 +135,19 @@ export default {
...
+          api.setAuthToken(loginToken);
           saveLoginToken(loginToken);
 
           store.dispatch.fetchUserInfo();
           store.dispatch.fetchWorkspaces();
 
-          const name = this.$route.name || 'home';
-          this.$router.replace({ name });
+          const newPath = this.$route.fullPath.slice(0, index);
+          this.$router.replace(newPath);
```
[the remark this answers is not in the document]
Comment (AlmightyYakob) [reply src/components/LoginMenu.vue]: Is this comment sufficient? It is placed on the function:

https://github.com/multinet-app/multinet-client/blob/5b5323d65451616311d19f15b1abade66a9c9426/src/components/LoginMenu.vue#L129-L134

If not, what more needs to be explained?

[the remark this answers is not in the document]
Comment (waxlamp) [reply src/components/LoginMenu.vue]: Not sure how I missed that before. Yes, that's sufficient. Thanks!

**Review by waxlamp — APPROVED** [review]

**Review by waxlamp — APPROVED** [review]

**Pull request merged**